mondrian/src/main/java/mondrian/olap/QueryTiming.java

```diff
@@ -47,7 +47,7 @@
      *
      * @param enabled Whether to collect stats in future
      */
-    public synchronized void init(boolean enabled) {
+    public void init(boolean enabled) {
```

Comment:
Was synchronization replaced by something else to keep the state consistent?

Reply:
It wasn't... I looked at how QueryTiming was used and basically the markStart, markEnd and markFull are the methods that could be called by different threads.  markStart and markEnd are always called by the RolapEvaluator's thread and I believe this is always done in a single thread.  Please correct me if I am wrong.  markFull is called when the SQL statement is closed so I suppose this could be called by a different thread than the RolapEvaluator's.  I can add the synchronization back to the markFull method.  

In general, there is overhead to making methods synchronized and markStart and markEnd are called A LOT regardless if profiling is turned on or not.

Reply:
I think you're right, the segment queries are definitely running on an Actor's spawn thread.

https://github.com/pentaho/mondrian/blob/master/mondrian/src/main/java/mondrian/rolap/SqlStatement.java#L314-L315

Also, when a cancelation comes in, it'll come from yet another thread (the user thread, or RolapResultShepherd) so that's another possible concurrent issue.

https://github.com/pentaho/mondrian/blob/master/mondrian/src/main/java/mondrian/rolap/RolapResultShepherd.java#L84

Reply:
@lucboudreau I added the synchronized keyword to markFull and also toString.  I also submitted a second commit with Sonar/Checkstyle fixes.  Please let me know if there are any other issues you see.  Otherwise, I think this is ready to be merged.  Thanks.